addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -172,38 +176,62 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 break;
             case CHANNEL_SWITCH:
                 if (command instanceof OnOffType) {
-                    fritzBox.setSwitch(ain, command.equals(OnOffType.ON) ? true : false);
+                    state.getSwitch().setState(OnOffType.ON.equals(command) ? SwitchModel.ON : SwitchModel.OFF);
+                    fritzBox.setSwitch(ain, OnOffType.ON.equals(command) ? true : false);
+                } else {
+                    logger.warn("Received unknown command {} for channel {}", command.toString(), CHANNEL_SWITCH);
                 }
                 break;
             case CHANNEL_SETTEMP:
                 if (command instanceof DecimalType) {
-                    BigDecimal temperature = new BigDecimal(command.toString());
+                    final BigDecimal temperature = new BigDecimal(command.toString());
+                    state.getHkr().setTist(temperature);
                     fritzBox.setSetTemp(ain, HeatingModel.fromCelsius(temperature));
+                    updateState(CHANNEL_RADIATOR_MODE, new StringType(state.getHkr().getRadiatorMode()));
+                } else if (command instanceof IncreaseDecreaseType) {
+                    final BigDecimal temperature = state.getHkr().getTsoll();
+                    if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                        temperature.add(BigDecimal.ONE);
+                    } else {
+                        temperature.subtract(BigDecimal.ONE);
+                    }
+                    state.getHkr().setTist(temperature);
+                    fritzBox.setSetTemp(ain, temperature);
+                    updateState(CHANNEL_RADIATOR_MODE, new StringType(state.getHkr().getRadiatorMode()));
                 } else if (command instanceof OnOffType) {
-                    BigDecimal temperature = command.equals(OnOffType.ON) ? HeatingModel.TEMP_FRITZ_ON
+                    final BigDecimal temperature = OnOffType.ON.equals(command) ? HeatingModel.TEMP_FRITZ_ON
                             : HeatingModel.TEMP_FRITZ_OFF;
+                    state.getHkr().setTist(temperature);
                     fritzBox.setSetTemp(ain, temperature);
+                    updateState(CHANNEL_RADIATOR_MODE, new StringType(state.getHkr().getRadiatorMode()));
+                } else {
+                    logger.warn("Received unknown command {} for channel {}", command.toString(), CHANNEL_SETTEMP);
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies.